### ISSUE TITLE
cron job

### ISSUE THREAD

**Issue by matador7495** [xcode75/XManagerPlus#152]
Even though smtp is working correctly, the emails that are waiting in the queue (queue_messages table in the database) are not being sent.
This problem existed before, I reinstalled the server, it was solved, and after a few days, the cron job had a problem again

Does anyone have a solution?

![image](https://user-images.githubusercontent.com/30972110/230768308-0ffd031b-1394-458f-8463-1fb3285ce23f.png)

This image is related to crontab -e

![image](https://user-images.githubusercontent.com/30972110/230768312-bdfd810b-2ca2-474c-8b3a-0cb25bf90634.png)

**Comment by amihos** [comment]
I have the same issue, no emails are being sent for data or account expiray.
![image](https://user-images.githubusercontent.com/35190548/233446846-a2c9571a-1237-43f5-baa4-8a164687edda.png)

**Comment by amihos** [comment]
@matador7495 I think I found something that might cause this issue. When the php xmplus queue is being run there is a lock file created called email_queue and its supposed to be deleted after job done.
Now if for any reason this process gets stuck and quits abnormally the file stays in this folder: /www/wwwroot/your_site_folder/storage 
so you need to manually delete it and then the php xmplus queue can run normally.
Now one of the reasons I think I found it might cause this is when you use google smtp and then you want to send many emails at the same time, like when you send notice to users. In this case I think google has a limit of sending emails at the same time and then it stops the process and the process gets into authentication error and stops abnormaly

**Comment by matador7495** [comment]
Yes, you are absolutely right, I noticed this file last week and the problem was solved by deleting it.
At the time of sending mass and consecutive messages, I think that there is an error due to the limitation of Google service smtp, and this causes the file not to be deleted.

The best solution is to detect if an error ever occurs, delete the email_queue file and try again in 1 hour.

@amihos @xcode75
